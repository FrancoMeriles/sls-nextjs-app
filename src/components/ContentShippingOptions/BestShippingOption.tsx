import { FC } from 'react'
import {
  Box,
  Heading,
  Button,
  Flex,
  Text,
  Modal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalBody,
  ModalCloseButton,
  useDisclosure,
  ModalFooter,
  chakra,
} from '@chakra-ui/react'

// Icons
import { Position, Truck } from '@base/icons'

// Utils
import { getPriceFormatted } from '@base/utils'

// Contexts
import { useZipCode } from '@base/contexts/ZipCode'

// Libraries
import dayjs from 'dayjs'
import 'dayjs/locale/es'
dayjs.locale('es')

interface BestShippingOptionProps {
  bestShippingOption: ShippingOption
  allwaysShippingOption: ShippingOption
}

interface ShippingOption {
  display: string
  shipping_option_type: string
  name: string
  cost: number
  base_cost: number
  estimated_delivery_time: {
    date: string
  }
}

const BestShippingOption: FC<BestShippingOptionProps> = ({
  bestShippingOption,
  allwaysShippingOption,
}) => {
  const zipCode = useZipCode()
  const { onOpen, isOpen, onClose } = useDisclosure()

  const bestArrive = dayjs(bestShippingOption.estimated_delivery_time.date).format(
    'dddd D [de] MMMM'
  )
  const agencyArrive = dayjs(allwaysShippingOption.estimated_delivery_time.date).format(
    'dddd D [de] MMMM'
  )

  return (
    <>
      <Flex m="15px 0">
        <Box w="30px" d="flex" alignItems="top" pt={1}>
          <Truck />
        </Box>
        <Box>
          {bestShippingOption.cost === 0 ? (
            <Text fontSize="sm" color="brand.200">
              Llega gratis <chakra.span fontWeight="semibold">el {bestArrive}</chakra.span>
            </Text>
          ) : (
            <Text fontSize="sm">
              Llega{' '}
              <chakra.span fontWeight="semibold" color="brand.200">
                el {bestArrive}
              </chakra.span>{' '}
              por {getPriceFormatted(bestShippingOption.base_cost)}
            </Text>
          )}
          <Text fontSize="xs" color="gray.500">
            Beneficio Mercado Puntos
          </Text>

          <Button onClick={onOpen} size="xs" colorScheme="brand" variant="link" mt={2}>
            Ver más formas de entrega
          </Button>
        </Box>
      </Flex>

      <Modal isOpen={isOpen} onClose={onClose} size="4xl" isCentered>
        <ModalOverlay />
        <ModalContent padding={6}>
          <ModalHeader>
            <Heading as="h3" size="lg">
              Opciones de envío y retiro
            </Heading>
          </ModalHeader>
          <ModalCloseButton color="brand.300" />
          <ModalBody p={8}>
            <Text fontSize="lg" fontWeight="semibold">
              Calculamos los costos y tiempos para esta dirección:
            </Text>
            <Flex marginTop={4} alignItems="center" p={5} bg="gray.100" borderRadius={5}>
              <Position color="brand.300" />
              <Text color="gray.500" ml={5} fontSize="sm">
                {zipCode && `CP: ${zipCode}`}
              </Text>
            </Flex>

            <Box marginTop={10} borderBottom="solid 1px #edf2f6">
              <Text fontSize="lg" fontWeight="semibold">
                Recibir compra
              </Text>
              <Flex justifyContent="space-between" p="30px 0">
                <Text>
                  Llega el <chakra.b>{bestArrive}</chakra.b> a tu dirección
                </Text>
                {bestShippingOption.cost === 0 ? (
                  <Text fontSize="md" color="brand.200">
                    Gratis
                  </Text>
                ) : (
                  <Text fontSize="md">{getPriceFormatted(bestShippingOption.base_cost)}</Text>
                )}
              </Flex>
            </Box>

            <Box marginTop={10} borderBottom="solid 1px #edf2f6">
              <Text fontSize="lg" fontWeight="semibold">
                Retirar compra
              </Text>
              <Flex justifyContent="space-between" p="30px 0">
                <Text>
                  Retiralo a partir del <chakra.b>{agencyArrive}</chakra.b> en correos
                </Text>
                {allwaysShippingOption.cost === 0 ? (
                  <Text fontSize="md" color="brand.200">
                    Gratis
                  </Text>
                ) : (
                  <Text fontSize="md">{getPriceFormatted(allwaysShippingOption.base_cost)}</Text>
                )}
              </Flex>
            </Box>
          </ModalBody>
          <ModalFooter>
            <Button variant="ghost" onClick={onClose}>
              Cerrar
            </Button>
          </ModalFooter>
        </ModalContent>
      </Modal>
    </>
  )
}

export default BestShippingOption
