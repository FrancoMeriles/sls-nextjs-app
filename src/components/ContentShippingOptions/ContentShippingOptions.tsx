import { FC } from 'react'
import {
  Box,
  Heading,
  Button,
  Flex,
  Text,
  Modal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalBody,
  ModalCloseButton,
  useDisclosure,
  ModalFooter,
  chakra,
} from '@chakra-ui/react'
import { Devolutions, Truck, Money, Calendar, Position } from '@base/icons'
import 'dayjs/locale/es'
import dayjs from 'dayjs'
import { getPriceFormatted } from '@base/utils'
import { useZipCode } from '@base/contexts/ZipCode'
dayjs.locale('es')

interface ContentShippingOptionsProps {
  options: OptionsProps
}

interface OptionsProps {
  find: any
  display: string
  shipping_option_type: string
  name: string
  cost: number
  base_cost: number
  estimated_delivery_time: {
    date: string
  }
}

const ContentShippingOptions: FC<ContentShippingOptionsProps> = ({ options }) => {
  const zipCode = useZipCode()
  const modalDevolution = useDisclosure()
  const modalDeliveryMethod = useDisclosure()
  const bestShippingOption = options.find(({ display }) => display === 'recommended')
  const allwaysShippingOption = options.find(
    ({ shipping_option_type }) => shipping_option_type === 'agency'
  )
  const bestArrive = dayjs(bestShippingOption.estimated_delivery_time.date).format(
    'dddd D [de] MMMM'
  )
  const agencyArrive = dayjs(allwaysShippingOption.estimated_delivery_time.date).format(
    'dddd D [de] MMMM'
  )
  return (
    <Box>
      <Flex m="15px 0">
        <Box w="30px" d="flex" alignItems="top" pt={1}>
          <Truck />
        </Box>
        <Box>
          {bestShippingOption.cost === 0 ? (
            <Text fontSize="sm" color="brand.200">
              Llega gratis <chakra.span fontWeight="semibold">el {bestArrive}</chakra.span>
            </Text>
          ) : (
            <Text fontSize="sm">
              Llega{' '}
              <chakra.span fontWeight="semibold" color="brand.200">
                el {bestArrive}
              </chakra.span>{' '}
              por {getPriceFormatted(bestShippingOption.base_cost)}
            </Text>
          )}
          <Text fontSize="xs" color="gray.500">
            Beneficio Mercado Puntos
          </Text>

          <Button
            onClick={modalDeliveryMethod.onOpen}
            size="xs"
            colorScheme="brand"
            variant="link"
            mt={2}
          >
            Ver más formas de entrega
          </Button>
        </Box>
      </Flex>
      <Flex m="15px 0">
        <Box w="30px" d="flex" alignItems="top" pt={1}>
          <Devolutions color="brand.200" />
        </Box>
        <Box>
          <Text color="brand.200" fontSize="sm">
            Devolución gratis
          </Text>
          <Text fontSize="xs" color="gray.500">
            Tenés 30 días desde que lo recibís.
          </Text>

          <Button
            onClick={modalDevolution.onOpen}
            size="xs"
            colorScheme="brand"
            variant="link"
            mt={2}
          >
            Conocer más
          </Button>
        </Box>
      </Flex>
      <Modal
        isOpen={modalDevolution.isOpen}
        onClose={modalDevolution.onClose}
        size="4xl"
        isCentered
      >
        <ModalOverlay />
        <ModalContent padding={6}>
          <ModalHeader>
            <Heading as="h3" size="lg">
              Podés devolver este producto gratis. ¡No importa el motivo!
            </Heading>
          </ModalHeader>
          <ModalCloseButton color="brand.300" />
          <ModalBody p={8}>
            <Flex marginTop={4}>
              <Flex w="60px" alignItems="center">
                <Calendar boxSize={7} />
              </Flex>
              <Box>
                <Text fontSize="md">Tenés 30 días desde que te llega.</Text>
              </Box>
            </Flex>
            <Flex marginTop={8}>
              <Flex w="60px" alignItems="center">
                <Devolutions boxSize={7} />
              </Flex>
              <Box>
                <Text fontSize="md">Podés enviarlo desde tu correo más cercano.</Text>
              </Box>
            </Flex>
            <Flex marginTop={8}>
              <Flex w="60px" alignItems="center">
                <Money boxSize={7} />
              </Flex>
              <Box>
                <Text fontSize="md">
                  Para devolverlo, el producto debe estar tal cual lo recibiste.
                </Text>
              </Box>
            </Flex>
          </ModalBody>
          <ModalFooter>
            <Button variant="ghost" onClick={modalDevolution.onClose}>
              Cerrar
            </Button>
          </ModalFooter>
        </ModalContent>
      </Modal>

      <Modal
        isOpen={modalDeliveryMethod.isOpen}
        onClose={modalDeliveryMethod.onClose}
        size="4xl"
        isCentered
      >
        <ModalOverlay />
        <ModalContent padding={6}>
          <ModalHeader>
            <Heading as="h3" size="lg">
              Opciones de envío y retiro
            </Heading>
          </ModalHeader>
          <ModalCloseButton color="brand.300" />
          <ModalBody p={8}>
            <Text fontSize="lg" fontWeight="semibold">
              Calculamos los costos y tiempos para esta dirección:
            </Text>
            <Flex marginTop={4} alignItems="center" p={5} bg="gray.100" borderRadius={5}>
              <Position color="brand.300" />
              <Text color="gray.500" ml={5} fontSize="sm">
                {zipCode && `CP: ${zipCode}`}
              </Text>
            </Flex>

            <Box marginTop={10} borderBottom="solid 1px #edf2f6">
              <Text fontSize="lg" fontWeight="semibold">
                Recibir compra
              </Text>
              <Flex justifyContent="space-between" p="30px 0">
                <Text>
                  Llega el <chakra.b>{bestArrive}</chakra.b> a tu dirección
                </Text>
                {bestShippingOption.cost === 0 ? (
                  <Text fontSize="md" color="brand.200">
                    Gratis
                  </Text>
                ) : (
                  <Text fontSize="md">{getPriceFormatted(bestShippingOption.base_cost)}</Text>
                )}
              </Flex>
            </Box>

            <Box marginTop={10} borderBottom="solid 1px #edf2f6">
              <Text fontSize="lg" fontWeight="semibold">
                Retirar compra
              </Text>
              <Flex justifyContent="space-between" p="30px 0">
                <Text>
                  Retiralo a partir del <chakra.b>{agencyArrive}</chakra.b> en correos
                </Text>
                {allwaysShippingOption.cost === 0 ? (
                  <Text fontSize="md" color="brand.200">
                    Gratis
                  </Text>
                ) : (
                  <Text fontSize="md">{getPriceFormatted(allwaysShippingOption.base_cost)}</Text>
                )}
              </Flex>
            </Box>
          </ModalBody>
          <ModalFooter>
            <Button variant="ghost" onClick={modalDeliveryMethod.onClose}>
              Cerrar
            </Button>
          </ModalFooter>
        </ModalContent>
      </Modal>
    </Box>
  )
}

export default ContentShippingOptions
