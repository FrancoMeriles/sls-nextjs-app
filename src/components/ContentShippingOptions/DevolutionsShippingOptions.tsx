import {
  Box,
  Heading,
  Button,
  Flex,
  Text,
  Modal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalBody,
  ModalCloseButton,
  useDisclosure,
  ModalFooter,
} from '@chakra-ui/react'

// Icons
import { Devolutions, Money, Calendar } from '@base/icons'

const DevolutionsShippingOptions = () => {
  const { onOpen, onClose, isOpen } = useDisclosure()

  return (
    <>
      <Flex m="15px 0">
        <Box w="30px" d="flex" alignItems="top" pt={1}>
          <Devolutions color="brand.200" />
        </Box>
        <Box>
          <Text color="brand.200" fontSize="sm">
            Devolución gratis
          </Text>
          <Text fontSize="xs" color="gray.500">
            Tenés 30 días desde que lo recibís.
          </Text>

          <Button onClick={onOpen} size="xs" colorScheme="brand" variant="link" mt={2}>
            Conocer más
          </Button>
        </Box>
      </Flex>
      <Modal isOpen={isOpen} onClose={onClose} size="4xl" isCentered>
        <ModalOverlay />
        <ModalContent padding={6}>
          <ModalHeader>
            <Heading as="h3" size="lg">
              Podés devolver este producto gratis. ¡No importa el motivo!
            </Heading>
          </ModalHeader>
          <ModalCloseButton color="brand.300" />
          <ModalBody p={8}>
            <Flex marginTop={4}>
              <Flex w="60px" alignItems="center">
                <Calendar boxSize={7} />
              </Flex>
              <Box>
                <Text fontSize="md">Tenés 30 días desde que te llega.</Text>
              </Box>
            </Flex>
            <Flex marginTop={8}>
              <Flex w="60px" alignItems="center">
                <Devolutions boxSize={7} />
              </Flex>
              <Box>
                <Text fontSize="md">Podés enviarlo desde tu correo más cercano.</Text>
              </Box>
            </Flex>
            <Flex marginTop={8}>
              <Flex w="60px" alignItems="center">
                <Money boxSize={7} />
              </Flex>
              <Box>
                <Text fontSize="md">
                  Para devolverlo, el producto debe estar tal cual lo recibiste.
                </Text>
              </Box>
            </Flex>
          </ModalBody>
          <ModalFooter>
            <Button variant="ghost" onClick={onClose}>
              Cerrar
            </Button>
          </ModalFooter>
        </ModalContent>
      </Modal>
    </>
  )
}

export default DevolutionsShippingOptions
