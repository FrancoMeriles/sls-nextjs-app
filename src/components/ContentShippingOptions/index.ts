import ContentShippingOptions from './ContentShippingOptions'

export default ContentShippingOptions
