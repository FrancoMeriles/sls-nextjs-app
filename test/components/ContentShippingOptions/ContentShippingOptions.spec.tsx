import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

import ContentShippingOptions from '@base/components/ContentShippingOptions'

const mockContentShippingOptionsProps = {
  options: [
    {
      display: 'recommended',
      shipping_option_type: '__SHIPPING_OPTIONS_TYPE_RC__',
      name: '__NAME_RC__',
      cost: 500,
      base_cost: 600,
      estimated_delivery_time: {
        date: '2021-08-03T00:00:00-03:00',
      },
    },
    {
      display: 'agency',
      shipping_option_type: 'agency',
      name: '__NAME_AG__',
      cost: 900,
      base_cost: 1000,
      estimated_delivery_time: {
        date: '2021-08-03T00:00:00-03:00',
      },
    },
  ],
}

describe('ContentShippingOptions', () => {
  it('should render and match snapshot with required props', () => {
    const tree = renderer
      .create(<ContentShippingOptions {...mockContentShippingOptionsProps} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should renders and display the correct data', async () => {
    render(<ContentShippingOptions {...mockContentShippingOptionsProps} />)
    expect(screen.getByText('Beneficio Mercado Puntos')).toBeInTheDocument()
  })
})
