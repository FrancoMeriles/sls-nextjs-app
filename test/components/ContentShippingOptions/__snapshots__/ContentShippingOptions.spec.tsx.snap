// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentShippingOptions should render and match snapshot with required props 1`] = `
<div
  className="css-0"
>
  <div
    className="css-1kowmw4"
  >
    <div
      className="css-1pla5ih"
    >
      <svg
        className="chakra-icon css-onkibi"
        focusable={false}
        viewBox="0 0 18 15"
      >
        <path
          d="M7.763 12.207a2.398 2.398 0 0 1-4.726 0H1.8a1.8 1.8 0 0 1-1.8-1.8V2.195a1.8 1.8 0 0 1 1.8-1.8h8.445a1.8 1.8 0 0 1 1.8 1.8v.568l3.322.035L18 6.821v5.386h-2.394a2.398 2.398 0 0 1-4.727 0H7.763zm-.1-1.2h3.182V2.195a.6.6 0 0 0-.6-.6H1.8a.6.6 0 0 0-.6.6v8.212a.6.6 0 0 0 .6.6h1.337a2.399 2.399 0 0 1 4.526 0zm7.843 0H16.8V7.179l-2.086-3.187-2.669-.029v5.76a2.399 2.399 0 0 1 3.461 1.284zm-2.263 1.99a1.198 1.198 0 1 0 0-2.395 1.198 1.198 0 0 0 0 2.396zm-7.843 0a1.198 1.198 0 1 0 0-2.395 1.198 1.198 0 0 0 0 2.396z"
          fill="currentColor"
          fillRule="nonzero"
        />
      </svg>
    </div>
    <div
      className="css-0"
    >
      <p
        className="chakra-text css-g5hezf"
      >
        Llega
         
        <span
          className="css-1ts511b"
        >
          el 
          martes 3 de agosto
        </span>
         
        por 
        $ 600,00
      </p>
      <p
        className="chakra-text css-1bmfkgj"
      >
        Beneficio Mercado Puntos
      </p>
      <button
        className="chakra-button css-1wd94o"
        onClick={[Function]}
        type="button"
      >
        Ver más formas de entrega
      </button>
    </div>
  </div>
  <div
    className="css-1kowmw4"
  >
    <div
      className="css-1pla5ih"
    >
      <svg
        className="chakra-icon css-me87sc"
        focusable={false}
        viewBox="0 0 14 12"
      >
        <path
          d="M2.474 7.2h7.225a2.7 2.7 0 1 0 0-5.4H7V.6h2.7a3.9 3.9 0 1 1 0 7.8H2.473l2.45 2.389-.839.859L.14 7.8l3.945-3.848.838.859L2.473 7.2z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      className="css-0"
    >
      <p
        className="chakra-text css-10wnejt"
      >
        Devolución gratis
      </p>
      <p
        className="chakra-text css-1bmfkgj"
      >
        Tenés 30 días desde que lo recibís.
      </p>
      <button
        className="chakra-button css-1wd94o"
        onClick={[Function]}
        type="button"
      >
        Conocer más
      </button>
    </div>
  </div>
</div>
`;
